x-pack/plugins/security_solution/server/lib/detection_engine/rule_types/threshold/find_threshold_signals.ts

```diff
@@ -142,30 +145,32 @@ export const findThresholdSignals = async ({
       secondaryTimestamp,
     });
 
-    const searchResultWithAggs = searchResult as ThresholdSingleBucketAggregationResult;
-    if (!searchResultWithAggs.aggregations) {
-      throw new Error('Aggregations were missing on threshold rule search result');
-    }
-
     searchAfterResults.searchDurations.push(searchDuration);
     searchAfterResults.searchErrors.push(...searchErrors);
 
-    const docCount = searchResultWithAggs.hits.total.value;
-    if (
-      docCount >= threshold.value &&
-      (!includeCardinalityFilter ||
-        (searchResultWithAggs.aggregations.cardinality_count?.value ?? 0) >=
-          threshold.cardinality[0].value)
+    if (searchResultHasAggs<ThresholdSingleBucketAggregationResult>(searchResult)) {
+      const docCount = searchResult.hits.total.value;
+      if (
+        docCount >= threshold.value &&
+        (!includeCardinalityFilter ||
+          (searchResult?.aggregations?.cardinality_count?.value ?? 0) >=
+            threshold.cardinality[0].value)
+      ) {
+        buckets.push({
+          doc_count: docCount,
+          key: {},
+          max_timestamp: searchResult.aggregations?.max_timestamp ?? { value: null },
+          min_timestamp: searchResult.aggregations?.min_timestamp ?? { value: null },
+          ...(includeCardinalityFilter
+            ? { cardinality_count: searchResult.aggregations?.cardinality_count }
+            : {}),
+        });
+      }
+    } else if (
+      !searchResultHasAggs<ThresholdSingleBucketAggregationResult>(searchResult) &&
```

Comment:
once you checked if response has aggregations in if condition before, do you need to do check here if it doesn't have aggregations? It looks like, this condition will be always `true`